contribs/sharing/src/main/java/org/matsim/contrib/sharing/logic/SharingLogic.java

```diff
@@ -143,8 +154,17 @@ public boolean tryPickupVehicle(double now, MobsimAgent agent) {
 		Activity pickupActivity = (Activity) plan.getPlanElements().get(pickupActivityIndex);
 		Verify.verify(pickupActivity.getType().equals(SharingUtils.PICKUP_ACTIVITY));
 
-		// Find closest vehicle and hope it is at the current station / link
-		Optional<VehicleInteractionPoint> selectedVehicleInteraction = service.findClosestVehicle(agent);
+		SharingVehicle reservedVehicle = service.hasReservationElseNull(agent);
```

Comment:
I would prefer Optional<SharingVehicle> getSharedVehicle(agent);

Reply:
changed to getReservedVehicle